File: labs/lab1/word_count/python/word_count.py
import os
import sys

def wc(content):
    return len(content.split())

def wc_file(filename):
    try:
        with open(filename, 'r', encoding='latin-1') as f:
            file_content = f.read()
        return wc(file_content)
    except FileNotFoundError:
        return 0

def wc_dir(dir_path, output_file):
    count = 0
    with open(output_file, 'a') as out_file:
        for filename in os.listdir(dir_path):
            filepath = os.path.join(dir_path, filename)
            if os.path.isfile(filepath):
                count += wc_file(filepath)
        out_file.write(str(count) + '\n')

def main():
    if len(sys.argv) != 3:
        print("Usage: python", sys.argv[0], "directory_path", "outputfile_path")
        return
    dir_path = os.path.abspath(sys.argv[1])
    output_file = os.path.abspath(sys.argv[2])
    wc_dir(dir_path, output_file)

if __name__ == "__main__":
    main()

File: labs/lab1/word_count/java/src/main/java/WordCount.java
import java.io.*;
import java.nio.file.*;
import java.util.*;

public class WordCount {

    public static void main(String[] args) {
        if (args.length != 2) {
            System.err.println("Usage: java WordCount <directory_path> <output_file_path>");
            System.exit(1);
        }

        String rootPath = args[0];
        String outputFilePath = args[1];
        wcDir(rootPath, outputFilePath);
    }

    public static int wc(String fileContent) {
        String[] words = fileContent.split("\\s+");
        return words.length;
    }

    public static int wcFile(String filename) {
        try {
            String fileContent = new String(Files.readAllBytes(Paths.get(filename)), "UTF-8");
            return wc(fileContent);
        } catch (IOException e) {
            return 0;
        }
    }

    public static void wcDir(String dirPath, String outputFilePath) {
        File output = new File(outputFilePath);
        int count = 0;

        try (FileWriter writer = new FileWriter(output, true)) {
            File dir = new File(dirPath);
            File[] files = dir.listFiles();

            if (files != null) {
                for (File file : files) {
                    if (file.isFile()) {
                        count += wcFile(file.getAbsolutePath());
                    }
                }
                writer.write(count + "\n");
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}
